collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -234,29 +233,33 @@ protected void goUpLevel() {
     }
 
     /**
-     * Builds a string representing the path of the current group. Each level is separated by `>`.
+     * Returns a string representing the 'path' of the current screen.
+     * Each level is separated by `>`.
      */
     private String getCurrentPath() {
         FormController formController = Collect.getInstance().getFormController();
         FormIndex index = formController.getFormIndex();
-        // move to enclosing group...
-        index = formController.stepIndexOut(index);
+
+        if (formController.getEvent(index) == FormEntryController.EVENT_QUESTION) {
+            // move to enclosing group...
+            index = formController.stepIndexOut(index);
+        }
 
         List<FormEntryCaption> groups = new ArrayList<>();
+
+        if (shouldShowRepeatGroupPicker()) {
+            groups.add(formController.getCaptionPrompt(repeatGroupPickerIndex));
+        }
+
         while (index != null) {
             groups.add(0, formController.getCaptionPrompt(index));
             index = formController.stepIndexOut(index);
         }
 
-        String path = ODKView.getGroupsPath(groups.toArray(new FormEntryCaption[groups.size()]));
+        // If the repeat picker is showing, don't show an item number for the current index.
+        boolean hideLastMultiplicity = shouldShowRepeatGroupPicker();
```

Comment:
I think this variable is redundant. The parameter in `getGroupsPath()` is named `hideLastMultiplicity` so ii think it's clear enough and you can call `shouldShowRepeatGroupPicker()` directly.

Reply:
I separated out this variable in order to explain why `hideLastMultiplicity` is used this way. If it's all squished into the return statement, it makes it less clear what the comment is referring to. I would prefer to keep it like this, do you have a strong preference?